### PR TITLE
Hotfix performance

### PR DESCRIPTION
**Pull request by jmgasper**

### CODE REVIEW

File: src/services/ProcessorService.js [hunk 1]
```diff
@@ -232,36 +232,49 @@ async function updateUserProfile (payload, connection) {
 
   // update the user email entry in the DB
   if (email !== undefined) {
-    await updateUserEmail(userId, email, connection)
+    // Only update the email if it's different than what's in the DB already
+    const currentEmailResult = await getEmailById(userId, connection)
+
+    logger.info("Query result:", JSON.stringify(currentEmailResult))
+    if(currentEmailResult!=null && currentEmailResult.length > 0){
```
Comment (testflyjets) [inline src/services/ProcessorService.js:239]: Please add spaces around `!=`

File: src/services/ProcessorService.js [hunk 2]
```diff
@@ -556,6 +569,21 @@ async function getUserCountById (userId, connection) {
   return connection.queryAsync('select count(*) count from user where user_id =' + userId)
 }
 
+/**
+ * Gets the user's current email address
+ * It is used to check if a user profile update event contains an updated email address to save
+ * or if the message can be ignored because the email address hasn't been updated.
+ *
+ * @param {String} userId  The user id to get the email address for
```
Comment (testflyjets) [inline src/services/ProcessorService.js:577]: Add a dash between the param name and the start of the description.

**Review by testflyjets — APPROVED** [review]
Just curious about all the commented-out lines.

File: src/services/ProcessorService.js [hunk 3]
```diff
@@ -232,36 +232,49 @@ async function updateUserProfile (payload, connection) {
...
+      else{
+        logger.info(`Skipping update to ${userId} because email hasn't changed from ${currentEmail}`)
+      }
+    }
   }
 
-  if (addresses !== undefined) {
-    await updateUserAddresses(payload, connection)
-  }
+  // if (addresses !== undefined) {
```
[the remark this answers is not in the document]
Comment (jmgasper) [reply src/services/ProcessorService.js:255]: Done.

File: src/services/ProcessorService.js (hunk 1 above)
Comment (jmgasper) [reply src/services/ProcessorService.js]: Done, also added comments about what's commented out

**Pull request merged**